logstash-core/lib/logstash/settings.rb

```diff
@@ -85,7 +85,7 @@ def registered?(setting_name)
 
     def get_setting(setting_name)
       setting = @settings[setting_name]
-      raise ArgumentError.new("Setting \"#{setting_name}\" hasn't been registered") if setting.nil?
+      raise ArgumentError.new("Setting \"#{setting_name}\" doesn't exist") if setting.nil?
```

Comment:
LGTM!

We could also be more informative, e.g.:

"doesn't exist (or hasn't been registered)."

Or to further that:

"doesn't exist (or hasn't been registered). Please check there are no typos in your setting."